src/jit/valuenum.cpp

```diff
@@ -1752,12 +1757,6 @@ ValueNum ValueNumStore::EvalFuncForConstantArgs(var_types typ, VNFunc func, Valu
         return EvalCastForConstantArgs(typ, func, arg0VN, arg1VN);
     }
 
-    if (typ == TYP_BYREF)
-    {
-        // We don't want to fold expressions that produce TYP_BYREF
-        return false;
```

Comment:
> Ah, so if Null is the VN for a zero constant value, why are these nodes getting the Null VN?

Because `EvalFuncForConstantArgs` returns `ValueNum` and `return false;` was casted to `ValueNum`(that is int), so this function returned `0(null)` for all byrefs with constant arguments.